src/Altinn.App.Core/Helpers/Serialization/ModelDeserializer.cs

```diff
@@ -61,50 +58,74 @@ public ModelDeserializer(ILogger logger, Type modelType)
                 return await DeserializeXmlAsync(stream);
             }
 
-            Error = $"Unknown content type {contentType}. Cannot read the data.";
-            return null;
+            return ModelDeserializerResult.FromError($"Unknown content type {contentType}. Cannot read the data.");
         }
 
-        private async Task<object?> DeserializeJsonAsync(Stream stream)
+        private async Task<ModelDeserializerResult> DeserializeMultipartAsync(Stream stream, string contentType)
         {
-            Error = null;
+            MediaTypeHeaderValue mediaType = MediaTypeHeaderValue.Parse(contentType);
+            string boundary = mediaType.Boundary.Value!.Trim('"');
+            var reader = new MultipartReader(boundary, stream);
+            MultipartSection? firstSection = await reader.ReadNextSectionAsync();
+            if (firstSection?.ContentDisposition?.Contains("name=\"dataModel\"") != true)
+            {
+                return ModelDeserializerResult.FromError("First entry in mulipart serialization must have name=\"dataModel\"");
+            }
+            var modelResult = await DeserializeJsonAsync(firstSection.Body);
+            if (modelResult.HasError)
+            {
+                return modelResult;
+            }
 
+            MultipartSection? secondSection = await reader.ReadNextSectionAsync();
+            Dictionary<string, string?>? reportedChanges = null;
+            if (secondSection is not null)
+            {
+                if (secondSection?.ContentDisposition?.Contains("name=\"previousValues\"") != true)
+                {
+                    return ModelDeserializerResult.FromError("First entry in mulipart serialization must have name=\"dataModel\"");
+                }
+                reportedChanges = await System.Text.Json.JsonSerializer.DeserializeAsync<Dictionary<string, string?>>(secondSection.Body);
+                if (await reader.ReadNextSectionAsync() != null)
+                {
+                    return ModelDeserializerResult.FromError("Multipart request had more than 2 elements");
```

Comment:
Should the error be even more specific and say that we expect "dataModel" and optionally a "previousValues" as we know that we only support these?

Reply:
Not sure if that is required. To get here we need the first element to be `dataModel` and the second to be `previousValues`, otherwise the earlier errors will trigger.

Reply:
thats true!